text/0000-node-api-v2.md

```diff
@@ -0,0 +1,1131 @@
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+## V1 Endpoints
+
+While the endpoints are documented in details [here](https://github.com/mimblewimble/grin/blob/master/doc/api/node_api.md), here is an overview of the REST API Endpoints of the v1 API.
+
+```JSON
+[
+  "get blocks",
+  "get headers",
+  "get chain",
+  "post chain/compact",
+  "get chain/validate",
+  "get chain/kernels/xxx?min_height=yyy&max_height=zzz",
+  "get chain/outputs/byids?id=xxx,yyy,zzz",
+  "get chain/outputs/byheight?start_height=101&end_height=200",
+  "get status",
+  "get txhashset/roots",
+  "get txhashset/lastoutputs?n=10",
+  "get txhashset/lastrangeproofs",
+  "get txhashset/lastkernels",
+  "get txhashset/outputs?start_index=1&max=100",
+  "get txhashset/merkleproof?n=1",
+  "get pool",
+  "post pool/push_tx",
+  "post peers/a.b.c.d:p/ban",
+  "post peers/a.b.c.d:p/unban",
+  "get peers/all",
+  "get peers/connected",
+  "get peers/a.b.c.d",
+  "get version"
+]
+```
+
+These endpoints can be grouped into 5 categories:
+
+- miscellaneous endpoints (which contain `status` and `version` endpoints)
+- `chain` endpoints (which also contain `blocks`, `chain` and headers` endpoints)
+- `peer` endpoints
+- `pool` endpoints
+- `txhashset` endpoints
+
+## V2 Endpoints
+
+The following endpoints are kept and refactored:
+
+```JSON
+[
+  "get blocks",
+  "get headers",
+  "get chain",
+  "post chain/compact",
+  "get chain/validate",
+  "get chain/kernels/xxx?min_height=yyy&max_height=zzz",
+  "get chain/outputs/byids?id=xxx,yyy,zzz",
+  "get chain/outputs/byheight?start_height=101&end_height=200",
+  "get status",
+  "get txhashset/outputs?start_index=1&max=100",
+  "get pool",
+  "post pool/push_tx",
+  "post peers/a.b.c.d:p/ban",
+  "post peers/a.b.c.d:p/unban",
+  "get peers/all",
+  "get peers/connected",
+  "get peers/a.b.c.d",
+  "get version"
+]
+```
+
+The logic of the following endpoints will NOT be implemented as they are purely internals:
+
+```JSON
+  "get txhashset/roots",
+  "get txhashset/lastoutputs?n=10",
+  "get txhashset/lastrangeproofs",
+  "get txhashset/lastkernels",
+  "get txhashset/merkleproof?n=1",
+```
+
+The new endpoint methods are the following:
+
+- [get_status](#get_status)
+- [get_version](#get_version)
+- [get_header](#get_header)
+- [get_block](#get_block)
+- [get_tip](#get_tip)
+- [get_kernel](#get_kernel)
+- [get_outputs](#get_outputs)
+- [get_unspent_outputs](#get_unspent_outputs)
+- [validate_chain](#validate_chain)
+- [compact_chain](#compact_chain)
+- [get_peers](#get_peers)
+- [get_connected_peers](#get_connected_peers)
+- [ban_peer](#ban_peer)
+- [unban_peer](#unban_peer)
+- [get_pool_size](#get_pool_size)
+- [get_stempool_size](#get_stempool_size)
+- [get_unconfirmed_transactions](#get_unconfirmed_transactions)
+- [push_transaction](#push_transaction)
+
+NB: The following v2 endpoints are classified by categories solely to simplify the reading and understanding of the RFC.
```

Comment:
It's not clear how it maps to physical endpoints (URL)

Reply:
Adding something about this.